x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/DriverContext.java

```diff
@@ -33,11 +35,30 @@
  */
 public class DriverContext {
 
+    /** A default driver context. The returned bigArrays is non recycling. */
+    public static DriverContext DEFAULT = new DriverContext(BigArrays.NON_RECYCLING_INSTANCE);
+
     // Working set. Only the thread executing the driver will update this set.
     Set<Releasable> workingSet = Collections.newSetFromMap(new IdentityHashMap<>());
 
     private final AtomicReference<Snapshot> snapshot = new AtomicReference<>();
 
+    private final BigArrays bigArrays;
+
+    // For testing
+    public DriverContext() {
+        this(BigArrays.NON_RECYCLING_INSTANCE);
+    }
```

Comment:
I know it's a pain, but I'd prefer not to have this in our production code. It'd be super ok on some test class, but it's too tempting.